### PR TITLE
flake.nix: Propagate nlohmann_json

### PR DESCRIPTION
**Pull request by roberth**
Without it, the nix headers are unusable.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -103,6 +102,7 @@
 
         propagatedDeps =
           [ (boehmgc.override { enableLargeConfig = true; })
+            nlohmann_json
```
Comment (regnat) [inline flake.nix:105]: Wouldn’t that also propagate it to the non-dev outputs? (Not that it’s a big dependency ofc, but still)

**Comment by edolstra** [comment]
Hm, but we're not using external nlohmann_json anymore (9f28dd97ae6afc68f0574a251325336c12d60c6f). I guess the problem is that we should install the headers in $dev/include/nix.

**Comment by roberth** [comment]
@edolstra Yes we are using external nlohmann_json. That commit didn't do much, but I think I've completed the removal of the external one in my second commit here.

File: src/libexpr/flake/lockfile.hh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 #include "flakeref.hh"
 
-#include <nlohmann/json_fwd.hpp>
+#include <nlohmann/json.hpp>
```
Comment (edolstra) [inline src/libexpr/flake/lockfile.hh:5]: No no, we should not include json.hpp in header files! IIRC it adds a huge amount of compilation time to every compilation unit.

**Comment by roberth** [comment]
Ok, the code base is in limbo between using an external nlohmann_json and a vendored nlohmann_json.
Clearly I don't know how it should be done, so feel free to take my commits and finish it. It shouldn't take much effort for someone who knows how it's best integrated.

**Pull request closed without merging**